src/raft.rs

```diff
@@ -625,20 +639,22 @@ impl<T: Storage> Raft<T> {
 
     /// Appends a slice of entries to the log. The entries are updated to match
     /// the current index and term.
-    pub fn append_entry(&mut self, es: &mut [Entry]) {
+    pub fn append_entry(&mut self, es: &mut [Entry]) -> Result<()> {
```

Comment:
Can you add a test that makes sure any future changes don't make it so the `Raft` is modified in an `append_entry` that results in an error?

Reply:
I can write a test that exercises this particular error condition, and makes sure it doesn't modify the raft log. There's no way to ensure that any hypothetical error conditions added in future changes also behave that way, since there's no way to ensure the test triggers those errors.

With this change, `append` and `maybe_append` both apply different checks and conditionally append to the log. It seems like it would be useful to more clearly document what their distinct purposes are.

Reply:
Improving the documentation would be good. :) Are you suggesting doing that in this PR or as future work?

Reply:
I don't know this code base well enough to do that. More of a suggestion for someone that does fully understand the intentions behind why those are separate.

Reply:
Ok! :ok_hand: I am still learning it too. Let's put this to the side for now.